src/scripts/router.js

```diff
@@ -3,31 +3,45 @@ import { renderer } from './renderer';
 import { camera } from './renderer/camera';
 import { PlanetsScene } from './scenes/planets';
 import { XrScene } from './scenes/xr-scene';
-
-// update scene when page loaded
-navigateToScene(window.location.pathname);
+import { PendulumScene } from './scenes/pendulums';
 
 /**
  * @type {XrScene}
  */
 let currentScene;
+const SavedStates = {};
+
+const Routes = {
+	get '/' () {
+		return new HomeScene(renderer, camera);
+	},
+	get '/planets' () {
+		return new PlanetsScene(renderer, camera);
+	},
+	get '/pendulums' () {
+		if (!this._cached_pendulum) {
+			this._cached_pendulum = new PendulumScene(renderer, camera);
+		}
+		this._cached_pendulum.isActive = true;
+		return this._cached_pendulum;
+	}
+};
 
 /**
  * update currently displayed scene based on the pathname
  * @param {string} pathname
  */
-function navigateToScene(pathname) {
+function navigateToScene(pathname, oldPath) {
   if (currentScene) {
     currentScene.isActive = false;
+    // Save the state from the previous scene
+    SavedStates[oldPath] = currentScene.state;
   }
 
-  switch (pathname) {
-    case '/':
-      currentScene = new HomeScene(renderer, camera);
-      break;
-    case '/planets':
-      currentScene = new PlanetsScene(renderer, camera);
-      break;
+  currentScene = (pathname in Routes) ? Routes[pathname] : Routes['/'];
+  if (pathname in SavedStates) {
```

Comment:
We should use the object built into the `window.history.pushState()` function to save state instead of implementing this ourselves

Reply:
That's a good point. I guess it makes sense to keep it the way you have it for now. 

@jonesjonathan brought up a good point that we might not really have any meaningful state to save, and on reload it will be gone anyway, so this might be overkill. 

My vote is to keep what you've done for now, though.